client/ayon_nuke/plugins/publish/validate_asset_context.py

```diff
@@ -111,4 +112,15 @@ def repair(cls, instance):
         for _key in invalid_keys:
             created_instance[_key] = instance.context.data[_key]
 
-        create_context.save_changes()
+        if instance.data["productType"] in {"prerender", "render", "image"}:
+            instance_node = instance.data["transientData"]["node"]
+            data = dict({
+                "folderPath": created_instance["folderPath"],
+                "task": created_instance["task"],
+                "productName": created_instance["productName"],
+                "productType": created_instance["productType"]
+            })
+
+            reset_write_node_filepath(instance_node, data)
+
+        create_context.save_changes()
```

Comment:
This _should_ be redundant - because the update instances of the Creator should already handle it.
```suggestion
        create_context.save_changes()

```

If it doesn't - we should debug in `update_instances` why it does not trigger what we need there. We should be able to rely on the creator class to update the instance.

Reply:
I have debugged and still dont understand why it's not working. It would be helpful if you can give some adivce.